editor/src/clj/editor/app_manifest.clj

```diff
@@ -0,0 +1,478 @@
+      {:foo true} map? {}
+      :foo number? 0
+      inc)
+    => {:foo 1}
+
+    (update-in-fixing
+      {:foo 1} map? {}
+      :foo number? 0
+      inc)
+    => {:foo 2}"
+  [m pred fix & args]
+  {:pre [(= 1 (rem (count args) 3))]}
+  (let [f-expr (last args)
+        forms (partition 3 (butlast args))
+        value-sym (gensym "value")
+        map-sym (gensym "map")]
+    `(let [~value-sym ~m
+           ~value-sym (if (not (~pred ~value-sym))
+                        ~fix
+                        ~value-sym)]
+      ~(reduce
+         (fn [form [k pred fix]]
+           `(let [~map-sym ~value-sym
+                  ~value-sym (get ~map-sym ~k ::not-found)
+                  ~value-sym (if (or (identical? ~value-sym ::not-found)
+                                     (not (~pred ~value-sym)))
+                               ~fix
+                               ~value-sym)]
+              (assoc ~map-sym ~k ~form)))
+         `(~f-expr ~value-sym)
+         (reverse forms)))))
+
+(defn get-toggle-value [manifest toggle]
+  (case (:toggle toggle)
+    :contains (let [{:keys [platform key value]} toggle]
+                (boolean (some #(= value %) (get-in-guarded manifest
+                                                            :platforms map?
+                                                            platform map?
+                                                            :context map?
+                                                            key vector?
+                                                            []))))
+    :boolean (let [{:keys [platform key value]} toggle
+                   default-value (not value)]
+               (= value (get-in-guarded manifest
+                                        :platforms map?
+                                        platform map?
+                                        :context map?
+                                        key boolean?
+                                        default-value)))))
+
+(defn set-toggle-value [manifest toggle value]
+  (case (:toggle toggle)
+    :contains (let [enabled value
+                    {:keys [platform key value]} toggle]
+                (update-in-fixing
+                  manifest map? {}
+                  :platforms map? {}
+                  platform map? {}
+                  :context map? {}
+                  key vector? []
+                  (if enabled
+                    (fn [values]
+                      (into [] (distinct) (conj (or values []) value)))
+                    (fn [values]
+                      (filterv #(not= value %) values)))))
+
+    :boolean (let [enabled value
+                   {:keys [platform key value]} toggle]
+               (update-in-fixing
+                 manifest map? {}
+                 :platforms map? {}
+                 platform map? {}
+                 :context map? {}
+                 key boolean? false
+                 (constantly (if enabled value (not value)))))))
```

Comment:
No, because the value might be `false`. The toggle "use android support lib" is enabled when `:jetifier` is set to `false`.

Reply:
But then `enabled` would be `false` and `(constantly enabled)` would work, right?

Reply:
No, `(constantly enabled)` wouldn't work, because when we enable the setting (e.g. enable "use android support lib"), we have `enabled` set to `true`, but the `jetifier` value in the manifest that corresponds to "enabled" state is `false`, i.e. `value` is `false`. It's confusing...